analyzers/src/SonarAnalyzer.VisualBasic/Rules/Hotspots/CreatingHashAlgorithms.cs

```diff
@@ -46,8 +46,8 @@ public CreatingHashAlgorithms()
         internal /*for testing*/ CreatingHashAlgorithms(IAnalyzerConfiguration analyzerConfiguration)
         {
             ObjectCreationTracker = new VisualBasicObjectCreationTracker(analyzerConfiguration, rule);
-            InvocationTracker = new VisualBasicInvocationTracker(analyzerConfiguration, rule);
-            BaseTypeTracker = new VisualBasicBaseTypeTracker(analyzerConfiguration, rule);
+            ParameterlessFactoryInvocationTracker = new VisualBasicInvocationTracker(analyzerConfiguration, rule);
+            ParameterizedFactoryInvocationTracker = new VisualBasicInvocationTracker(analyzerConfiguration, rule);
```

Comment:
This is suspicious. Tracker instance doesn't persist any state about tracked methods and can be reused. One tracker of given type should be just fine. We can call `Track` multiple times with different conditions.
```suggestion
            InvocationTracker = new VisualBasicInvocationTracker(analyzerConfiguration, rule);
```